azure/functions/decorators/function_app.py

```diff
@@ -0,0 +1,967 @@
+        """Get dictionary representation of the bindings of the function.
+
+        :return: Dictionary representation of the bindings.
+        """
+        stub_bindings_f_json: Dict[str, List[Dict]] = {"bindings": []}
+        for b in self._bindings:
+            stub_bindings_f_json["bindings"].append(b.get_dict_repr())
+        return stub_bindings_f_json
+
+    def get_dict_repr(self) -> Dict:
+        """Get the dictionary representation of the function.
+
+        :return: The dictionary representation of the function.
+        """
+        stub_f_json = {
+            "scriptFile": self.function_script_file
+        }
+        stub_f_json.update(self.get_bindings_dict())  # NoQA
+        return stub_f_json
+
+    def get_user_function(self) -> Callable:
+        """Get the python function customer defined.
+
+        :return: The python function customer defined.
+        """
+        return self._func
+
+    def get_function_name(self) -> str:
+        """Get the function name.
+
+        :return: Function name.
+        """
+        return self._name
+
+    def get_function_json(self) -> str:
+        """Get the json stringified form of function.
+
+        :return: The json stringified form of function.
+        """
+        return json.dumps(self.get_dict_repr())
+
+    def __str__(self):
+        return self.get_function_json()
+
+
+class FunctionBuilder(object):
+    def __init__(self, func, function_script_file):
+        self._function = Function(func, function_script_file)
+
+    def __call__(self, *args, **kwargs):
+        pass
+
+    def configure_function_name(self, function_name: str) -> 'FunctionBuilder':
+        self._function.set_function_name(function_name)
+
+        return self
+
+    def add_trigger(self, trigger: Trigger) -> 'FunctionBuilder':
+        self._function.add_trigger(trigger=trigger)
+        return self
+
+    def add_binding(self, binding: Binding) -> 'FunctionBuilder':
+        self._function.add_binding(binding=binding)
+        return self
+
+    def _validate_function(self) -> None:
+        function_name = self._function.get_function_name()
+        trigger = self._function.get_trigger()
+        if trigger is None:
+            raise ValueError(
+                f"Function {function_name} does not have a trigger")
+
+        bindings = self._function.get_bindings()
+        if trigger not in bindings:
+            raise ValueError(
+                f"Function {function_name} trigger {trigger} not present"
+                f" in bindings {bindings}")
+
+        if isinstance(trigger, HttpTrigger) and trigger.route is None:
+            trigger.route = self._function.get_function_name()
+
+    def build(self) -> Function:
+        self._validate_function()
+        return self._function
+
+
+class FunctionsApp:
+    """FunctionsApp object used by worker function indexing model captures
+    user defined functions and metadata.
+
+    Ref: https://aka.ms/azure-function-ref
```

Comment:
Should this be pointing to Python-dev reference guide of the general dev reference guide?